### PR TITLE
Fixed: failing test if exe is missing

### LINKED ISSUE #203: Pull Sonarr commit 'Added update check early in startup if the package requested a post-install update check'
Pull https://github.com/Sonarr/Sonarr/commit/25abf52b3fe5f4befeff9d35128d0c03e533a373

```
git cherry-pick 25abf52b3fe5f4befeff9d35128d0c03e533a373
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### LINKED ISSUE #859: Pull Sonarr commit 'Added mechanism for package maintainers to produce a health check error.'
Pull https://github.com/Sonarr/Sonarr/commit/7da02c236aa03e6aef011130526040c1cb8399fc

```
git cherry-pick -ex 7da02c236aa03e6aef011130526040c1cb8399fc
```

This commit has 3 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

[AB#726](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/726)

### LINKED ISSUE #860: Pull Sonarr commit 'Fixed failing test'
Pull https://github.com/Sonarr/Sonarr/commit/4c324fbbbfc761d7aab459ba70059ba95ab1b286

```
git cherry-pick -ex 4c324fbbbfc761d7aab459ba70059ba95ab1b286
```

This commit has 0 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

[AB#727](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/727)

### PR DESCRIPTION
**Pull request by PearsonFlyer**
#### Database Migration
NO

#### Description
Properly fails update test if the Readarr.Update.exe is missing.
Adds mechanism for package maintainers to produce a health check error (required for above fix).

#### Todos
- [ ] Tests
- [ ] Wiki Updates

#### Issues Fixed or Closed by this PR

* Fixes #860 
* Fixes [AB#727](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/727)
* Fixes #859
* Fixes [AB#726](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/726)
* Fixes #203

### CODE REVIEW

**Comment by ta264** [comment]
This probably needs Sonarr -> Radarr

File: src/NzbDrone.Core.Test/UpdateTests/UpdatePackageProviderFixture.cs [hunk 1]
```diff
@@ -33,13 +33,14 @@ public void finds_update_when_version_lower()
         }
 
         [Test]
-        [Ignore("Ignore until we actually release something on Master")]
+        [Ignore("Ignore until we actually release something on Main")]
```
Comment (Qstick) [inline src/NzbDrone.Core.Test/UpdateTests/UpdatePackageProviderFixture.cs:36]: Master

File: src/NzbDrone.Core/Configuration/ConfigFileProvider.cs [hunk 2]
```diff
@@ -175,7 +175,7 @@ public AuthenticationType AuthenticationMethod
 
         public bool AnalyticsEnabled => GetValueBoolean("AnalyticsEnabled", true, persist: false);
 
-        public string Branch => GetValue("Branch", "master").ToLowerInvariant();
+        public string Branch => GetValue("Branch", "main").ToLowerInvariant();
```
Comment (Qstick) [inline src/NzbDrone.Core/Configuration/ConfigFileProvider.cs:178]: Master

File: src/NzbDrone.Core/Configuration/DeploymentInfoProvider.cs [hunk 3]
```diff
@@ -33,14 +38,15 @@ public DeploymentInfoProvider(IAppFolderInfo appFolderInfo, IDiskProvider diskPr
             var releaseInfoPath = Path.Combine(bin, "release_info");
 
             PackageUpdateMechanism = UpdateMechanism.BuiltIn;
-            DefaultBranch = "aphrodite";
+            DefaultBranch = "main";
```
Comment (Qstick) [inline src/NzbDrone.Core/Configuration/DeploymentInfoProvider.cs:41]: Master

File: src/NzbDrone.Core/Update/InstallUpdateService.cs [hunk 4]
```diff
@@ -218,18 +227,13 @@ private UpdatePackage GetUpdatePackage(CommandTrigger updateTrigger)
...
-            }
-
             if (OsInfo.IsNotWindows && !_configFileProvider.UpdateAutomatically && updateTrigger != CommandTrigger.Manual)
             {
-                _logger.ProgressDebug("Auto-update not enabled, not installing available update.");
+                _logger.ProgressDebug("Auto-update not enabled, not installing available update");
                 return null;
             }
 
+
```
Comment (Qstick) [inline src/NzbDrone.Core/Update/InstallUpdateService.cs:235]: No double blank

File: src/NzbDrone.Core/Update/InstallUpdateService.cs [hunk 5]
```diff
@@ -281,5 +285,65 @@ public void Execute(ApplicationUpdateCommand message)
                 }
             }
         }
+
+        public void Handle(ApplicationStartingEvent message)
```
Comment (Qstick) [inline src/NzbDrone.Core/Update/InstallUpdateService.cs:289]: Need to add the file for ApplicationStartingEvent

**Review by Qstick — CHANGES_REQUESTED** [review]

**Comment by bakerboy448** [comment]
@PearsonFlyer wrong branch pushed here?

**Pull request closed without merging**